### PR TITLE
[US-152] [US-153]Allow no cmap for truetype fonts.

### PR DESCRIPTION
**Pull request by anovik**
The aim of this PR is to address [US-152] and [US-153]. Previously we could not use embedded fonts without cmap but it turns out we can use some of them, with Identity-H or Identity-V encodings. 

For this purpose, if for TrueType font we haven't found cmap, we don't return an error now, but instead raise hasNoCmap flag. 

On unipdf side, we check if this flag is raised for identity encoded embedded font - we use it and do not search for a replacement. It is useful mostly for Asian fonts.

Another minor change is to allow hhea length > 36 for TrueType. They still can't have hhea length < 36.

### CODE REVIEW

File: truetype/truetype.go [hunk 1]
```diff
@@ -533,6 +544,11 @@ func (f *Font) HasShortCmap() bool {
 	return f.hasShortCmap
 }
 
+// HasNoCmap returns true if font doesn't have cmap. This may be acceptable for Identity-H or Identity-V encoded fonts.
+func (f *Font) HasNoCmap() bool {
+	return f.hasNoCmap
+}
```
Comment (sampila) [inline truetype/truetype.go]: Probably better to rename it as `HasCmap`?

**Review by sampila — CHANGES_REQUESTED** [review]
Overall looks good, added a comment

File: truetype/truetype.go (hunk 1 above)
Comment (anovik) [reply truetype/truetype.go]: @sampila Changed to HasCmap

**Review by sampila — APPROVED** [review]
Looks good to me

**Review by gunnsth — APPROVED** [review]
LGTM

**Pull request merged**